src/java.base/share/classes/sun/security/jca/ProviderList.java

```diff
@@ -399,8 +399,8 @@ public Service getService(String type, String name) {
      *
      * The List returned is NOT thread safe.
      */
-    public List<Service> getServices(String type, String algorithm) {
-        return new ServiceList(type, algorithm);
+    public Iterator<Service> getServices(String type, String algorithm) {
```

Comment:
Change the javadoc of this method from "Return a List of all ..." to "Returns an iterator over all ..." and replace other instances of "List" with "Iterator".